ecommerce/src/main/java/app/controller/HomeController.java

```diff
@@ -1,17 +1,23 @@
 package app.controller;
 
+import java.util.List;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 
 import app.authentication.MyDBAuthenticationService;
+import app.model.Category;
+import app.service.Impl.CategoryServiceImpl;
 
 @Controller
 public class HomeController {
 	@Autowired
 	private MyDBAuthenticationService authenticationService;
+	@Autowired
+	private CategoryServiceImpl categoryService;
```

Comment:
```suggestion
	private CategoryService categoryService;
```
nếu em dùng class thì nó đâu có nghĩa j nữa em